src/GafferImage/ImageWriter.cpp

```diff
@@ -2069,6 +2109,10 @@ void ImageWriter::execute() const
 		{
 			part.spec.attribute("view", part.views[0] );
 		}
+		else if( forceNukeViewMetadata )
+		{
+			part.spec.attribute("view", "main" );
+		}
```

Comment:
Since this is the only place we use `forceNukeViewMetadata`, perhaps we could move the evaluation of `layoutPartNamePlug()` here, to limit the circumstances in which we use the funky context? If we did that, could we even use `part.channels[0]` to provide valid values for the context variables other than `imageWriter:nukeViewName`?

Reply:
So, this sounds like you want to do multiple queries to decide forceNukeViewMetadata, rather than just evaluating the plug once?

If that's the case, then maybe it makes sense to just make `partChannelNameDataType` do 2 evaluations of `layoutPartNamePlug()` every time, and return an additional boolean in the tuple for whether we need to engage weirdness. The name of this function is getting pretty unwieldy, so we'd probably rename it to something like: evaluateLayoutForChannel.

Then it can use the actual channel name in the context. We probably want the actual view metadata behaviour to be global, so we'd enable it if any channel name triggers something to write __nukeViewProbe to the part name.